### ISSUE TITLE
The downloaded video has no sound

### ISSUE THREAD

**Issue by 4strodev** [kkdai/youtube#342]
I tried it with different videos and all has the same problem. Maybe it's a failure that only happens on Linux. I found this similar issue #279, but the provided advice doesn't work.

**Comment by noahlove** [comment]
Not all streams have audio. Are you sure you are downloading one that has audio?

**Comment by 4strodev** [comment]
Yes I tried with three different videos all of them with audio. I don't know if the bug is something replicable or not, but I just cloned the repo and executed the CLI, giving them a YouTube link. The video downloaded successfully but without audio.

**Comment by 7z3m** [comment]
same happened to me as well.

**Comment by Mrkouhadi** [comment]
I have the same issue. 
I kinda solved it, but i don't get the best quality, i always get either 360p or something lower even though the video has multiple options for quality.

`	
    var format *youtube.Format
    for _, f := range video.Formats {
		    if f.AudioChannels > 0 && f.QualityLabel != "" { // Ensure it has both video and audio
			    if format == nil || f.Height > format.Height {
				    format = &f // Select the format with the highest resolution
			    }
		    }
    }
`

**Comment by 7z3m** [comment]
indeed, i share the low quality issue with you. thanks for sharing the code.

**Comment by keepfocused** [comment]
same problem. 
link: https://www.youtube.com/watch?v=nxWqANttAdA
command line: /Users/myUser/go/bin/youtubedr download https://www.youtube.com/watch\?v\=nxWqANttAdA 
no sound 😥
